OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -197,6 +198,12 @@ void IRenderAboveWorld.RenderAboveWorld(Actor self, WorldRenderer wr)
 			var viewport = new Rectangle(center - new int2(viewportSize) / 2, viewportSize);
 			var wcr = Game.Renderer.WorldRgbaColorRenderer;
 
+			// SwingSpeed is defined in px/tick so we must account for the fraction of a tick that elapsed since the last render.
+			// The scale is capped at 1 tick to avoid unexpected behaviour at game start, if RunTime overflows, or if the game stalls.
+			var runtime = Game.RunTime;
+			var tickFraction = Math.Min((runtime - lastRender) * 1f / world.Timestep, 1);
```

Comment:
`tickFraction` looks like something we might want to eventually calculate and expose at a central, publicly accessible place like `Game`, so we can reuse it for things like interpolation and don't need to do the math locally in every place that needs it.

Just thinking aloud, not a request for this particular PR.

Reply:
You are thinking of something like https://love2d.org/wiki/dt delta time?